src/test/scala/iog/psg/cardano/util/InMemoryCardanoApi.scala

```diff
@@ -32,7 +32,11 @@ trait InMemoryCardanoApi {
   }
 
   implicit final class InMemoryExecutor[T](req: CardanoApiRequest[T]) {
-    def executeOrFail(): T = inMemoryExecutor.execute(req).futureValue.getOrElse(fail("Request failed."))
+    def executeOrFail(): T =
+      inMemoryExecutor.execute(req).futureValue match {
+        case Left(value)  => fail("Request failed: " + value.message)
+        case Right(value) => value
+      }
```

Comment:
:heart_eyes: style much more readable than on lhs.

Best to get out of the habit of using java string concatenation (+) and use s"${}"
It's a little more error prone and purists get worked up.